Mathlib/FieldTheory/LinearDisjoint.lean

```diff
@@ -0,0 +1,362 @@
+
+- `IntermediateField.LinearDisjoint.of_basis_mul`:
+  `IntermediateField.LinearDisjoint.of_basis_mul'`:
+  conversely, if `{ a_i }` is an `F`-basis of `A`, if `{ b_j }` is an `F`-basis of `L`,
+  such that the family `{ a_i * b_j }` in `E` is `F`-linearly independent,
+  then `A` and `L` are linearly disjoint.
+
+Other main results:
+
+- `IntermediateField.LinearDisjoint.symm`, `IntermediateField.linearDisjoint_symm`:
+  linearly disjoint is symmetric.
+
+- `IntermediateField.LinearDisjoint.rank_sup_of_isAlgebraic`,
+  `IntermediateField.LinearDisjoint.finrank_sup`:
+  if `A` and `B` are linearly disjoint,
+  then the rank of `A ⊔ B` is equal to the product of the rank of `A` and `B`.
+
+  **TODO:** remove the algebraic assumptions (the proof becomes complicated).
+
+- `IntermediateField.LinearDisjoint.of_finrank_sup`:
+  conversely, if `A` and `B` are finite extensions,
+  such that rank of `A ⊔ B` is equal to the product of the rank of `A` and `B`,
+  then `A` and `B` are linearly disjoint.
+
+- `IntermediateField.LinearDisjoint.of_finrank_coprime`:
+  if the rank of `A` and `B` are coprime,
+  then `A` and `B` are linearly disjoint.
+
+- `IntermediateField.LinearDisjoint.inf_eq_bot`:
+  if `A` and `B` are linearly disjoint, then they are disjoint.
+
+## Linear disjoint abstract fields
+
+We adapt the definitions in <https://mathoverflow.net/questions/8324>.
+
+### Main definitions
+
+- `Field.LinearDisjoint`: for two abstract fields `E` and `K` over `F`, they are
+  linearly disjoint over `F` (`Field.LinearDisjoint F E K`), if `E ⊗[F] K` is a field.
+
+  In this case, it can be shown that at least one of `E / F` and `K / F` are algebraic, and if this
+  holds, then it is equivalent to the above `IntermediateField.LinearDisjoint`.
+
+The advantage of `Field.LinearDisjoint` is that it is preserved under algebra isomorphisms.
+This is not true for `IntermediateField.LinearDisjoint` if both of the
+extensions are not algebraic.
+
+### Main results
+
+- **TODO:** if `Field.LinearDisjoint F E K`,
+  then at least one of `E / F` and `K / F` are algebraic.
+
+- **TODO:** if at least one of `E / F` and `K / F` are algebraic,
+  then `Field.LinearDisjoint` is equivalent to `IntermediateField.LinearDisjoint`.
+
+## Tags
+
+linearly disjoint, linearly independent, tensor product
+
+-/
+
+open scoped Classical TensorProduct
+
+open Module IntermediateField
+
+noncomputable section
+
+universe u v w
+
+namespace IntermediateField
+
+variable {F : Type u} {E : Type v} [Field F] [Field E] [Algebra F E]
+
+variable (A B : IntermediateField F E)
+
+variable (L : Type w) [Field L] [Algebra F L] [Algebra L E] [IsScalarTower F L E]
+
+/-- If `A` is an intermediate field of `E / F`, and `E / L / F` is a field extension tower,
+then `A` and `L` are linearly disjoint, if they are linearly disjoint as subalgebras of `E`
+(`Subalgebra.LinearDisjoint`). -/
+protected abbrev LinearDisjoint : Prop :=
+  A.toSubalgebra.LinearDisjoint (IsScalarTower.toAlgHom F L E).range
```

Comment:
Then the dot notation cannot be used anymore since it does not have a parameter of type `IntermediateField`.

Originally both of the parameters are `IntermediateField` but later it seems that it's probably useful (but probably not) if one of the parameter can be generalized to an abstract field.

Reply:
Alright! Can you please add a short note explaining this in a new section `### Implementation notes` in the module docstring?

Reply:
Done.